src/main/java/dev/blaauwendraad/masker/json/ByteTrie.java

```diff
@@ -85,67 +102,108 @@ public void insert(String word) {
                         1. Locale issues (see String#equalsIgnoreCase)
                         2. So we don't have to convert when searching
                      */
-                    node.children[lowerBytes[i] + MAX_BYTE_SIZE] = child;
-                    node.children[upperBytes[i] + MAX_BYTE_SIZE] = child;
+                    node.children[lowerBytes[i] + BYTE_OFFSET] = child;
+                    node.children[upperBytes[i] + BYTE_OFFSET] = child;
                 }
             }
             node = child;
         }
+        node.keyMaskingConfig = config.getConfig(word);
```

Comment:
Nice, we can use it to check if the key should be interpreted as a jsonpath to fix the bug you mentioned yesterday.